### DISCUSSION TITLE
How suspend query that force-refetches not-stale data on mount

### DISCUSSION THREAD

**Discussion by chekrd** [TanStack/query#4099] [Q&A]
Hi, is there a solution for a situation where:
- an entity is cached, staletime is 10 minutes, (consider the entity to be fresh in this situation - the next action happens within 10 minutes)
- an editor of the entity is mounted and we need to force refetch the query and block the editor from rendering until done (suspense mode enabled)

I didn't find a solution to this. All I tried still renders with the entity from the cache and then rerenders after the refetch (if changed).
The entity is rarely updated but in the editor, we want to be sure we have the latest state on the first render in a specific component.

**Comment by TkDodo** [comment]
you can always check for `isFetching` and disable / unmount the editor then?
